### ISSUE TITLE
Contains over `filter{}.count > 0` 

### ISSUE THREAD

**Issue by dirtydanee** [realm/SwiftLint#2803]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

On collections we can filter the elements based on a predicate, count them and compare against zero to check if there is at least one element satisfying that predicate. However, we can use the `contain` API to achieve the same result.

What should trigger?
```swift
struct Foo {
    let bar: Bool
}

let foos = [Foo(bar: false), Foo(bar: true)]
let canBar: Bool = foos.filter { $0.bar }.count > 0
```

```swift
struct Foo {
    let bar: Bool
}

let foos = ["1": Foo(bar: false),"2": Foo(bar: true)]
let canBar = foos.filter { $1.bar }.count > 0
```

```swift
let canNumber2 = [1, 2, 3].filter { $0 == 2 }.count > 0
```

```swift
let numbers: Set<Int> = [1, 2, 3]
numbers.filter { $0 == 2 }.count > 0
```

What should not trigger?

```swift
struct Foo {
    let bar: Bool
}

let foos = [Foo(bar: false), Foo(bar: true)]
let canBar: Bool = foos.contains(where: { $0.bar })
```

```swift
struct Foo {
    let bar: Bool
}

let foos = ["1": Foo(bar: false),"2": Foo(bar: true)]
let canBar = foos.contains { $1.bar }
```

```swift
let canNumber2 = [1, 2, 3].contains(2)
```

```swift
let numbers: Set<Int> = [1, 2, 3]
numbers.contains(2)
```

The rules should not be configurable and could be enabled by default.

**Comment by idrougge** [comment]
`empty_count` would already catch this, but a separate rule might offer better fixits.
